libs/contrib/Text/Parser/Core.idr

```diff
@@ -109,6 +109,20 @@ export
        Grammar tok (c1 || c2) b
 (*>) x y = map (const id) x <*> y
 
+||| Produce a grammar that can parse a different type of token by providing a
+||| function converting the new token type into the original one.
+export
+mapToken : (a -> b) -> Grammar b c ty -> Grammar a c ty
```

Comment:
Dubious name choice . Other ideas for consideration: `tmap`, `lmap`, `contramap`.

Reply:
name is fine.

Reply:
`contralmapToken` ? :)

Reply:
`hocusTokens`